torch/utils/data/datapipes/iter/selecting.py

```diff
@@ -12,28 +12,54 @@ class FilterIterDataPipe(MapIterDataPipe):
 
     Iterable DataPipe to filter elements from datapipe according to filter_fn.
     args:
-        datapipe: Iterable DataPipe being filterd
+        datapipe: Iterable DataPipe being filtered
         filter_fn: Customized function mapping an element to a boolean.
         fn_args: Positional arguments for `filter_fn`
         fn_kwargs: Keyword arguments for `filter_fn`
+        drop_empty_batches: By default, drops batch if it is empty after filtering instead of keeping an empty list
+        nesting_level: Determines which level the fn gets applied to, by default it applies to the top level (= 0)
     """
     def __init__(self,
                  datapipe: IterDataPipe[T_co],
                  filter_fn: Callable[..., bool],
                  fn_args: Optional[Tuple] = None,
                  fn_kwargs: Optional[Dict] = None,
+                 drop_empty_batches: bool = True,
+                 nesting_level: int = 0,
                  ) -> None:
-        super().__init__(datapipe, fn=filter_fn, fn_args=fn_args, fn_kwargs=fn_kwargs)
+        self.drop_empty_batches = drop_empty_batches
```

Comment:
This is a typed class definition, so you should declare `drop_empty_batches: bool` inside of the class

Reply:
To make sure I'm understanding correctly.
I should add `bool` as a type to `self.drop_empty_batches` like so?
```
self.drop_empty_batches: bool =  drop_empty_batches
```

Reply:
no, you do it like this:

```
class A:
  drop_empty_batches: bool
  # your other stuff
```